### PR TITLE
Migrate [SearchUser]

### PR DESCRIPTION
**Pull request by geoseong**
#22 
I figured out testing `Context API`.
> I can't figure out the way how to find `Context API` 's objects. :(
but I tried hard to make coverage high lol

### CODE REVIEW

File: src/components/screen/SearchUser.js [hunk 1]
```diff
@@ -102,107 +103,101 @@ const StyledSearchImage = styled.Image`
...
+  };
+  const renderItem = ({ item, index }: { item: Friend, index: number}) => {
     return (
       <UserListItem
-        id='user'
+        testID={`userListItem${index}`}
+        testObj={{ profileModal }}
         user={item}
-        onPress={() => this.showProfileModal(item, data)}
+        onPress={(e) => userListOnPress(item)}
```
Comment (hyochan) [inline src/components/screen/SearchUser.js]: How to utilize `e`?

File: src/components/screen/SearchUser.js [hunk 2]
```diff
@@ -102,107 +103,101 @@ const StyledSearchImage = styled.Image`
...
-  renderItem = (item: Friend, data: any) => {
+  const userListOnPress = (item: Friend) => {
+    if (profileModal && profileModal.dispatch) {
+      profileModal.dispatch({
+        type: 'show-modal',
+        payload: { user: item, deleteMode: true },
+      });
+    }
+  };
+  const renderItem = ({ item, index }: { item: Friend, index: number}) => {
```
Comment (hyochan) [inline src/components/screen/SearchUser.js:127]: Why not just
```
const renderItem = ({ item: Friend, index: numnber }) => {
```

?

**Review by hyochan — CHANGES_REQUESTED** [review]
You are almost there. Please see my few comments!

File: src/components/screen/SearchUser.js (hunk 1 above)
Comment (geoseong) [reply src/components/screen/SearchUser.js]: I just unconsciously added `e` as a param. `e` is used nowhere... do you want me to remove this parameter `e`?

File: src/components/screen/SearchUser.js (hunk 2 above)
Comment (geoseong) [reply src/components/screen/SearchUser.js:127]: what you've suggested is not working.
when I did this, It throws error. error message is like `'index' is not defined` | `'item' is not defined` in eslint.
This is `destructuring` so I think defining as Flowtype in destructuring is not working.
plz refer to this link -> https://github.com/facebook/flow/issues/235#issuecomment-72592788

File: src/components/screen/SearchUser.js (hunk 1 above)
Comment (hyochan) [reply src/components/screen/SearchUser.js]: I think it is better to remove it if not used.

File: src/components/screen/SearchUser.js (hunk 2 above)
Comment (hyochan) [reply src/components/screen/SearchUser.js:127]: Ok this looks new to me. Thank you for detail!

File: src/components/screen/SearchUser.js (hunk 1 above)
Comment (geoseong) [reply src/components/screen/SearchUser.js]: I removed it. [0539db6](https://github.com/dooboolab/talktalk-rn/pull/44/commits/0539db6ba5c0ac56117f4738226d08ba12bfe7f6)

**Comment by geoseong** [comment]
I resolved what you've adviced and figured out how to test `Context API on React Hooks`.
I don't think it is the best way but it can track `state` of `Context` injecting `context object` into `props ` on `Component`.

**Review by hyochan — APPROVED** [review]
Great work!

**Pull request merged**